tests/integration/test_schema_registry_auth.py

```diff
@@ -0,0 +1,215 @@
+    assert res.status_code == 200
+    assert [1] == res.json()
+
+
+async def test_sr_ids(registry_async_client_auth: Client) -> None:
+    cavesubject = new_random_name("cave-")
+    carpetsubject = new_random_name("carpet-")
+
+    res = await registry_async_client_auth.post(
+        f"subjects/{quote(cavesubject)}/versions", json={"schema": schema_avro_json}, auth=aladdin
+    )
+    assert res.status_code == 200
+    avro_sc_id = res.json()["id"]
+    assert avro_sc_id >= 0
+
+    res = await registry_async_client_auth.post(
+        f"subjects/{quote(carpetsubject)}/versions",
+        json={"schemaType": "JSON", "schema": schema_jsonschema_json},
+        auth=admin,
+    )
+    assert res.status_code == 200
+    jsonschema_sc_id = res.json()["id"]
+    assert jsonschema_sc_id >= 0
+
+    res = await registry_async_client_auth.get(f"schemas/ids/{avro_sc_id}", auth=aladdin)
+    assert res.status_code == 200
+
+    res = await registry_async_client_auth.get(f"schemas/ids/{jsonschema_sc_id}", auth=aladdin)
+    assert res.status_code == 404
+    assert {"error_code": 40403, "message": "Schema not found"} == res.json()
+
+    res = await registry_async_client_auth.get(f"schemas/ids/{avro_sc_id}", auth=reader)
+    assert res.status_code == 404
+    assert {"error_code": 40403, "message": "Schema not found"} == res.json()
+
+    res = await registry_async_client_auth.get(f"schemas/ids/{jsonschema_sc_id}", auth=reader)
+    assert res.status_code == 200
+
+
+async def test_sr_auth_forwarding(registry_async_auth_pair: List[str]) -> None:
+    auth = requests.auth.HTTPBasicAuth("admin", "admin")
+
+    # Test primary/replica forwarding with global config setting
+    primary_url, replica_url = registry_async_auth_pair
+    max_tries, counter = 5, 0
+    wait_time = 0.5
+    for compat in ["FULL", "BACKWARD", "FORWARD", "NONE"]:
+        resp = requests.put(f"{replica_url}/config", json={"compatibility": compat}, auth=auth)
+        assert resp.ok
+        while True:
+            if counter >= max_tries:
+                raise Exception("Compat update not propagated")
+            resp = requests.get(f"{primary_url}/config", auth=auth)
+            if not resp.ok:
+                print(f"Invalid http status code: {resp.status_code}")
+                continue
+            data = resp.json()
+            if "compatibilityLevel" not in data:
+                print(f"Invalid response: {data}")
+                counter += 1
+                await asyncio.sleep(wait_time)
+                continue
+            if data["compatibilityLevel"] != compat:
+                print(f"Bad compatibility: {data}")
+                counter += 1
+                await asyncio.sleep(wait_time)
+                continue
+            break
+
+
+async def test_rest_api_with_sr_auth(rest_async_client_registry_auth: Client, admin_client: KafkaRestAdminClient) -> None:
```

Comment:
Just checking: this tests that we can produce via Kafka REST without credentials when SR auth is enabled, right?

Reply:
Added comment why this test exists